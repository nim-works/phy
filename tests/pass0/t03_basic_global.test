discard """
  output: "(Done 1)"
"""
(TypeDefs
  (Int 8)
  (ProcTy (Type 0))
)
(GlobalDefs
  (GlobalDef (Int 8) (IntVal 1))
)
(ProcDefs
  (ProcDef (Type 1) (Locals) (Continuations
    (Continuation (Params) 0
      (Continue 1 (Copy (Global 0)))
    )
    (Continuation (Params (Type 0)))
  ))
)