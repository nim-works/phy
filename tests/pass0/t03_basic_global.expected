.type t0 (Int)
.type t1 (Proc t0)
.global g0 1
.start t1 p0
  GetGlobal g0
  Ret
.end
